reVX/least_cost_xmission/cost_creator.py

```diff
@@ -0,0 +1,410 @@
+        base_line_costs_fpath : None | str
+            Base line cost per mile for ISOs file. Use default if None
+        iso_lookup_fpath : None | str
+            JSON file linking ISO region code to name. Use default if None
+        power_classes_fpath : None | str
+            JSON of line capacities to use for reV power classes. Use default
+            if None
+        sloper_lyr : str
+            Name of slope layer in input_h5_fpath
+        ncld_lyr : str
+            Name of NLCD (land use) layer in input_h5_fpath
+        land_use_classes : dict | None
+            NCLD land use codes corresponding to use classes for multipliers.
+            If None use defaults.
+        default_mults : None | dict
+            Multipliers for regions not specified in iso_mults_fpath. Use
+            defaults if None. Note that there are no default line costs.
+        tiff_dir : str | None
+            Path to save costs and intermediary rasters as geotiffs in. Don't
+            save to geotiff if None.
+        cell_size : int
+            Raster cell size for all layers (meters). Assumed to be square.
+        """
+        # TODO - verify profile for iso_regions_fpath and input_h5_fpath are
+        # comparable.
+
+        xc = XmissionConfig(iso_mults_fpath=iso_mults_fpath,
+                            base_line_costs_fpath=base_line_costs_fpath,
+                            iso_lookup_fpath=iso_lookup_fpath,
+                            power_classes_fpath=power_classes_fpath)
+
+        xcc = cls(h5_fpath, iso_regions_fpath, xc['iso_lookup'])
+
+        if tiff_dir is None:
+            save_geotiff = False
+        else:
+            save_geotiff = True
+
+        if land_use_classes is None:
+            land_use_classes = NLCD_LAND_USE_CLASSES
+
+        mults_arr = xcc.compute_multipliers(input_h5_fpath, slope_lyr,
+                                            nlcd_lyr, land_use_classes,
+                                            xc['iso_mults'], default_mults)
+
+        if save_geotiff:
+            xcc.create_geotiff(tiff_dir, 'multipliers.tif', mults_arr)
+
+        for power_class, capacity in xc['power_classes'].items():
+            logger.info(f'Calculating costs for class {power_class} using a '
+                        f'{capacity}MW line')
+            blc_arr = xcc.compute_base_line_costs(capacity,
+                                                  xc['base_line_costs'],
+                                                  cell_size)
+
+            if save_geotiff:
+                xcc.create_geotiff(tiff_dir, f'blc_{capacity}MW.tif', blc_arr)
+
+            costs_arr = blc_arr * mults_arr
+
+            xcc.save_layer(f'tie_line_costs_{capacity}MW', costs_arr)
+
+            if save_geotiff:
+                xcc.create_geotiff(tiff_dir, f'tie_line_costs{capacity}MW.tif',
+                                   costs_arr)
+
+    @classmethod
+    def build_test_costs(cls, h5_fpath, iso_regions_fpath, input_h5_fpath,
```

Comment:
This isn't useful outside of the tests, we should just move this code to the tests as a function that can be run if needed.